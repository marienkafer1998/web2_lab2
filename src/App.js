import React from "react";
import Geolocation from "./components/Geolocation";
import Favorites from "./components/Favorites";


class App extends React.Component {
    render() {
        return(
            <div>
                <Geolocation />
                <Favorites />
            </div>
        )

    }

}

export default App;

