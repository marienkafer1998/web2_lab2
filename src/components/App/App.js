import React from "react";
import Geolocation from "../Geolocation/Geolocation";
import Favorites from "../Favorites/Favorites";


class App extends React.Component {
    render() {
        return(
            <div>
                <Geolocation />
                <Favorites />
            </div>
        )
    }
}

export default App;

