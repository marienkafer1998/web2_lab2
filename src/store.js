import { createStore, combineReducers, applyMiddleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { LOCAL_STORAGE_KEY } from "./LocalStorage";

import geoReducer from "./redux/geoReducer";
import favReducer from "./redux/favReducer";

const store = createStore(
    combineReducers({geo: geoReducer, fav: favReducer}),
    applyMiddleware(logger, thunk)
);



store.subscribe(() => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify([...store.getState().fav.favorites.keys()]));
});

export default store;
