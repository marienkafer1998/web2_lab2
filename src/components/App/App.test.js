import React from 'react';
import renderer from 'react-test-renderer';
import App from "./App.js";
import {Provider} from "react-redux";
import store from "../../store";

it('App functional rendered properly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <App/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

