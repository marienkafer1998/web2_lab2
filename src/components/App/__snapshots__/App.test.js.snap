// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App functional rendered properly 1`] = `
<div>
  <div
    className="weather container"
  >
    <div
      className="container"
    >
      <div
        className="col"
      >
        Weather is here
      </div>
      <div
        className="col"
      >
        <button
          aria-label={null}
          className="btn btn-secondary btn-lg"
          onClick={[Function]}
          type="button"
        >
          Get geolocation
        </button>
        <div
          className="container"
        >
          <h1>
            Please, wait, weather is loading...
          </h1>
          <img
            alt="Weather is loading"
            src="https://anatomised.com/wp-content/uploads/2016/05/spinner-test4.gif"
            width="100px"
          />
        </div>
        <div>
          Error: there is no geolocation
        </div>
      </div>
    </div>
  </div>
  <div
    className="favorite-weather container"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h1
            className="fav"
          >
            Favorites
          </h1>
        </div>
        <div
          className="col-sm-5 offset-sm-1"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              className="adder"
              name="cityName"
              placeholder="Add new city"
              required={true}
              text-transform="capitalize"
              type="text"
            />
            <button
              aria-label={null}
              className="btn btn-secondary"
              onClick={[Function]}
              type="submit"
            >
              +
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      className="row"
    />
  </div>
</div>
`;
